examples/with-electron/renderer/pages/index.js

```diff
@@ -1,54 +1,47 @@
-import { Component } from 'react'
+import { useState, useEffect } from 'react'
 
-export default class Home extends Component {
-  state = {
-    input: '',
-    message: null,
-  }
+const Home = () => {
+  const [input, setInput] = useState('')
+  const [message, setMessage] = useState(null)
 
-  componentDidMount() {
-    // start listening the channel message
-    global.ipcRenderer.on('message', this.handleMessage)
-  }
+  useEffect(() => {
+    global.ipcRenderer.on('message', (event, message) => setMessage(message))
 
-  componentWillUnmount() {
-    // stop listening the channel message
-    global.ipcRenderer.removeListener('message', this.handleMessage)
-  }
-
-  handleMessage = (event, message) => {
-    // receive a message from the main process and save it in the local state
-    this.setState({ message })
-  }
+    return () => {
+      global.ipcRenderer.removeListener('message', (event, message) =>
+        setMessage(message)
+      )
```

Comment:
The `removeListener` here will not remove the added listener since the function reference isn't the same. You can fix it by storing the function in a variable:
```js
useEffect(() => {
  const handleMessage = (event, message) => setMessage(message);
  global.ipcRenderer.on('message', handleMessage)

  return () => {
    global.ipcRenderer.removeListener('message', handleMessage);
  }
}, [])
```

Reply:
good catch @rafaelalmeidatk , resolved. 🚀